File: app/src/main/java/com/keelim/cnubus/ui/Setting/SettingFragment.java
package com.keelim.cnubus.ui.Setting;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.keelim.cnubus.R;
import com.keelim.cnubus.activities.DeveloperActivity;

public class SettingFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.settings_preferences);
    }

    @Override
    public boolean onPreferenceTreeClick(Preference preference) {
        String key = preference.getKey();
        if (key.equals("developer")) {
            Intent intent_developer = new Intent(getContext(), DeveloperActivity.class);
            startActivity(intent_developer);
            return true;
        }
        return false;
    }
}

File: features/ui-setting/src/main/java/com/keelim/ui_setting/SettingFragment.kt
package com.keelim.ui_setting

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.ui.platform.LocalContext
import androidx.fragment.app.Fragment
import com.keelim.common.startActivity
import com.keelim.compose.ui.setThemeContent
import com.keelim.ui_setting.ui.Section
import com.keelim.ui_setting.ui.SettingScreen
import okhttp3.Route

class SettingFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return setThemeContent {
            val context = LocalContext.current

            SettingScreen { action ->
                val route = when(action){
                    Section.Developer -> Section.Developer
                }

                context.startActivity<SettingActivity>(
                    "type" to route
                )
            }
        }
    }
}